### PR TITLE
Calculate sdk tags

### PR DESCRIPTION
**Pull request by sigurdm**

### CODE REVIEW

File: lib/src/package_analyzer.dart [hunk 1]
```diff
@@ -7,6 +7,7 @@ import 'dart:collection';
 import 'dart:convert';
 import 'dart:io';
 
+import 'package:async/async.dart';
```
Comment (isoos) [inline lib/src/package_analyzer.dart]: do you need this?

File: lib/src/pubspec.dart [hunk 2]
```diff
@@ -112,6 +112,21 @@ class Pubspec {
 
   SdkConstraintStatus get sdkConstraintStatus =>
       SdkConstraintStatus.fromSdkVersion(_inner.environment['sdk']);
+
+  /// The sdk tags associated with [this].
+  List<String> sdkTags() {
+    final result = <String>[];
+    final dependentSdks = this.dependentSdks;
+    if (dependentSdks.every(['dart'].contains)) {
```
Comment (isoos) [inline lib/src/pubspec.dart]: I find this hard to parse. Isn't this the same as:
```dart
if (dependentSdks.length == 1 && dependentSdks.single =='dart') {
```

File: lib/src/pubspec.dart [hunk 3]
```diff
@@ -112,6 +112,21 @@ class Pubspec {
...
+      result.add('sdk:dart');
+    }
+    if (dependentSdks.every(['dart', 'flutter'].contains)) {
```
Comment (isoos) [inline lib/src/pubspec.dart]: Same here:
```dart
if (dependentSdks.length == 2 && dependentSdks.containsAll(['dart', 'flutter']) {
```

File: lib/src/package_analyzer.dart (hunk 1 above)
Comment (sigurdm) [reply lib/src/package_analyzer.dart]: Nope - removed

File: lib/src/pubspec.dart (hunk 2 above)
Comment (sigurdm) [reply lib/src/pubspec.dart]: Yes, I wrote it like that initially, but changed it to follow the same pattern as below.

File: lib/src/pubspec.dart (hunk 3 above)
Comment (sigurdm) [reply lib/src/pubspec.dart]: No - not exactly the same.
the expression i wrote is true for `dependentSdks = ['dart']` (You get the 'sdk:flutter' tag as soon as you can be used in flutter, not if you require flutter).
I agree that it is cryptic, and would be happy to write it in a better way.

Comment (isoos) [reply lib/src/pubspec.dart]: I see, makes sense. Please add a code comment with this information, so that we'll remember it a few months down the line :).

Comment (sigurdm) [reply lib/src/pubspec.dart]: Good idea! Thanks
